scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -561,6 +564,41 @@ static u64 sched_prio_to_latency_weight(u64 prio)
 	return sched_prio_to_weight[DL_MAX_LAT_PRIO - prio - 1];
 }
 
+static void do_iowait_boost(struct task_struct *p, struct task_ctx *taskc, s32 cpu)
+{
+	if (!iowait_boost || !p || !taskc || cpu < 0)
+		return;
+	if (!(p->flags & PF_KTHREAD) || (p->flags & TASK_IDLE))
+		return;
+
+	if (BPF_CORE_READ_BITFIELD_PROBED(p, in_iowait)) {
+		u64 now = bpf_ktime_get_ns();
+		if (taskc->boost_ctx.deadline == 0)
+			taskc->boost_ctx.deadline = now +
+				iowait_boost_max_dur_ms * NSEC_PER_MSEC;
+
+		if (now > taskc->boost_ctx.deadline) {
+			taskc->boost_ctx.curr_ios = 0;
+			taskc->boost_ctx.deadline = 0;
+			return;
+		}
+
+		taskc->boost_ctx.curr_ios++;
+		if (taskc->boost_ctx.curr_ios <= iowait_boost_max_ios*2) {
+			if (taskc->boost_ctx.level == 0)
+				taskc->boost_ctx.level = SCX_CPUPERF_ONE;
+			__COMPAT_scx_bpf_cpuperf_set(cpu, min(taskc->boost_ctx.level, SCX_CPUPERF_ONE));
```

Comment:
So probably a dumb question, but don't we need to also _lower_ the cpuperf setting for other cores in order for this to have an effect? We initialize the cpuperf value to `SCX_CPUPERF_ONE` [here](https://github.com/sched-ext/sched_ext/blob/sched_ext/kernel/sched/ext.c#L4945-L4946) when the scheduler is loaded, so I'm not quite following how this is changing anything, but I'm probably missing something.